src/httpClient.ts

```diff
@@ -160,32 +172,29 @@ class DefaultHTTPClient implements HTTPClient {
       }
       response_start = performance.now();
 
-      if (!error && response.status >= 400) {
+      if (!error && response.statusCode >= 400) {
         let message: string | undefined = undefined;
-        if (response.headers.get("Content-Type") === "application/json") {
+        let { body_text } = await consumeResponseText(response);
+        if (response.headers["content-type"] === "application/json") {
           try {
-            const body = (await response.json()) as any;
+            const body = JSON.parse(body_text) as any;
             if (body && body.status === "error") {
               message = body.error;
             } else {
-              message = JSON.stringify(body);
+              message = body_text;
             }
           } catch (_: unknown) {
             /* empty */
           }
         } else {
-          try {
-            const body = await response.text();
-            if (body) {
-              message = body;
-            }
-          } catch (_: unknown) {
-            /* empty */
-          }
+          message = body_text;
         }
-        error = new TurbopufferError(message ?? response.statusText, {
-          status: response.status,
-        });
+        error = new TurbopufferError(
+          message ?? `http error ${response.statusCode}`,
+          {
+            status: response.statusCode,
+          },
+        );
       }
       if (
         error &&
```

Comment:
Might be worth adding the retries to the requestTiming, to ensure those are extremely infrequent (as they'll add a lot of latency). A `retryTime` timing, for example

Reply:
for a followup, I think we're adding enough metrics rn, and we know the issue isn't retries because we see all the time in the old body_read_time metric